js/components/closed-caption-multi-audio-menu/helpers.js

```diff
@@ -57,31 +57,72 @@ function getDisplayLanguage(languagesList, languageCode) {
 
 /**
  * Gets display title based on language and label
- * @param {String} language - language string attribute
- * @param {String} label - label string attribute
- * @param {String} noLanguageText - label for a case when
+ * @param {Object} trackParam - AudioTrack params
+ * @param {String} trackParam.language - language string attribute
+ * @param {String} trackParam.label - label string attribute
+ * @param {String} trackParam.langCode - code of the language
+ * @param {String} trackParam.noLanguageText - label for a case when
  * we do not have values for language and label of an audioTrack
  * @returns {String} displayTitle - human readable display title
  */
-function getDisplayTitle(language, label, noLanguageText) {
+function getDisplayTitle(trackParam) {
   // set default function params
-  var displayLanguage = language || '';
-  var displayLabel = label || '';
+  var displayLanguage = trackParam.language || '';
+  var displayLabel = trackParam.label || '';
+  var noLanguageText = trackParam.noLanguageText || CONSTANTS.SKIN_TEXT.UNDEFINED_LANGUAGE;
+  var langCode = trackParam.langCode;
 
   if (!displayLanguage.length && !displayLabel.length) {
-    if (typeof noLanguageText === 'undefined' || !noLanguageText.length) {
-      noLanguageText = CONSTANTS.SKIN_TEXT.UNDEFINED_LANGUAGE;
-    }
     return noLanguageText;
   } else if (displayLanguage.length && !displayLabel.length) {
     return displayLanguage;
   } else if (!displayLanguage.length && displayLabel.length) {
     return displayLabel;
   } else {
-    return displayLanguage.concat(' ', displayLabel);
+    if (langCode === CONSTANTS.LANGUAGE.UNDEFINED_LANGUAGE) {
+      displayLanguage = '';
+    }
+    return displayLanguage.concat(' ', displayLabel).trim();
   }
 }
 
+/**
+ * If language in audioTrack is special we need to use localized value if it is possible.
+ * We can localize the value using audio code.
+ * @param {String} langCode - code of the language
+ * @param {String} userLanguage - language which user would like to use
+ * @param {Object} localizableStrings - mapping of string keys to localized values
+ * @param {Object} languageMap - object with codes of special languages as keys
+ * and labels for the codes as values
+ * @returns {String} localized special audio track name or original name or empty string
+ */
+function getLocalizedSpecialLanguage(langCode, userLanguage, localizableStrings, languageMap) {
+  var phrase = languageMap ? languageMap[langCode] : '';
+  var localizedLanguage = Utils.getLocalizedString(
+    userLanguage,
+    phrase,
+    localizableStrings
+  );
+  localizedLanguage = localizedLanguage ?
+    localizedLanguage
+    :
+    phrase ? phrase : '';
+  return localizedLanguage;
+}
+
+/**
+ *
```

Comment:
Can we add a description for this function?

Reply:
\+